### PR TITLE
feat: add semgrep action

### PR DESCRIPTION
**Pull request by chengjoey**
## Description
add semgrep action

## Checklist

 - [ ] I made sure to check the compatibility of the erda version statemented in the action's spec.yml.
 - [ ] My change is adequately tested.

### CODE REVIEW

File: actions/semgrep/1.0/spec.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    schema: http
+locale:
+  zh-CN:
+    displayName: Semgrep 代码扫描
+    desc: 用于对代码仓库执行代码质量检测，生成质量检测报告
+    params.code_dir.desc: 执行代码质量分析的目录
+    parmas.config.desc: 扫描的规则
+    parmas.format.desc: 报告的格式，可选值：json、sarif
+  en-US:
+    displayName: Sonar Code Quality Check
```
Comment (Effet) [inline actions/semgrep/1.0/spec.yml:36]: bad name

**Review by Effet — CHANGES_REQUESTED** [review]

File: actions/semgrep/1.0/spec.yml (hunk 1 above)
Comment (chengjoey) [reply actions/semgrep/1.0/spec.yml]: adjusted

desc: Code Analysis at Ludicrous Speed, find bugs, run security scans in CI, and enforce security standards across your organization.

File: actions/semgrep/1.0/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      params:
+        # 代码扫描目录
+        code_dir: ${git-checkout}
+        # 代码扫描规则，可以是本地文件或者线上规则
+        config: p/ci
+        # 报告输出格式
+        format: sarif
+        # 可选参数
+        args:
+          - --error
```
Comment (Effet) [inline actions/semgrep/1.0/README.md:19]: given the best simple practice, and append a more scene-related case. please do not mix all option in one example.

**Review by Effet — CHANGES_REQUESTED** [review]

File: actions/semgrep/1.0/README.md (hunk 2 above)
Comment (chengjoey) [reply actions/semgrep/1.0/README.md]: add two best practrice
1. use auto rule
2. custom rule file

Comment (chengjoey) [reply actions/semgrep/1.0/README.md]: add introduction to common optional parameters

**Review by Effet — APPROVED** [review]

**Pull request merged**